flang/include/flang/Parser/parse-tree.h

```diff
@@ -4169,6 +4199,15 @@ struct OmpSimpleStandaloneDirective {
   CharBlock source;
 };
 
+// Ref: [5.1:217-220], [5.2:291-292]
+//
+// interop -> INTEROP clause[ [ [,] clause]...]
+struct OpenMPInteropConstruct {
+  TUPLE_CLASS_BOILERPLATE(OpenMPInteropConstruct);
+  CharBlock source;
```

Comment:
The `source` values here gets updated when you pass the parser result into sourced function in `openmp-parsers.cpp`.

So, check if the same is possible for `InteropType`, `InteropPreference` and `OmpInitClause`.

Reply:
Hi @Thirumalai-Shaktivel , 

I found the root cause of the issue. The reason why the source information is not carried back to Interop Construct is because we have check all the semantic check inside the Interop Construct block on ```check-omp-structures.cpp``` and not on individual clauses. 

So instead of using ```GetContext().clauseSource```, if we use ```GetContext().directiveSource```, we are getting the source information for the error. Hence, the semantic checks are now passing with ```test_errors.py```. 

Addressed the issue on the commit [#9b0cd4b](https://github.com/swatheesh-mcw/llvm-project/pull/1/commits/9b0cd4b2c15449788bb1b66a4832d9c3be8c4195).